ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpEngine.kt

```diff
@@ -22,7 +22,7 @@ class OkHttpEngine(
     override val config: OkHttpConfig
 ) : HttpClientJvmEngine("ktor-okhttp"), WebSocketEngine {
 
-    private val engine = OkHttpClient.Builder()
+    private val engine = config.preconfigured ?: OkHttpClient.Builder()
         .apply(config.config)
```

Comment:
Doesn't it cause `config.preconfigured` to be mutated/polluted? Will the preconfigured instance accumulate configs? What if we create several clients concurrently with the same preconfigured instance?